backend/penndata/management/commands/get_venture_events.py

```diff
@@ -0,0 +1,112 @@
+
+        try:
+            resp = requests.get(VENTURE_EVENTS_WEBSITE, headers=HEADERS)
+        except ConnectionError:
+            print("Error:", ConnectionError)
+            return None
+
+        soup = BeautifulSoup(resp.text, "html.parser")
+
+        event_containers = soup.find_all("div", class_="PromoSearchResultEvent")
+        last_start_datetime = None
+
+        for event in event_containers:
+            event_date_elem = event.find("div", class_="PromoSearchResultEvent-eventDate")
+            event_start_datetime = None
+            event_end_datetime = None
+            # some events don't have a start/end date time or a year
+            if event_date_elem:
+                event_date_str = event_date_elem.text.strip()
+
+                event_date_parts = event_date_str.split(" at ")
+                event_start_str = event_date_parts[1].split(" - ")[0].strip()
+                event_end_str = event_date_parts[1].split(" - ")[1].strip()
+
+                event_start_datetime = datetime.strptime(
+                    event_date_parts[0] + " " + event_start_str, "%B %d, %Y %I:%M%p"
+                )
+                event_end_datetime = datetime.strptime(
+                    event_date_parts[0] + " " + event_end_str, "%B %d, %Y %I:%M%p"
+                )
+                last_start_datetime = event_start_datetime
+            else:  # if no year given
+                event_month_elem = event.find("div", class_="PromoSearchResultEvent-month")
+                event_day_elem = event.find("div", class_="PromoSearchResultEvent-day")
+
+                if event_month_elem and event_day_elem:
+                    event_month = event_month_elem.text.strip()
+                    event_day = int(event_day_elem.text.strip())
+
+                    if last_start_datetime:  # has to be before any previous events
+                        if datetime.strptime(event_month, "%B").month > last_start_datetime.month:
+                            start_year = current_year - 1
+                        else:
+                            start_year = current_year
+                    else:  # if no date time yet
+                        # if in future and next year
+                        if current_month > datetime.strptime(event_month, "%B").month:
+                            start_year = current_year + 1
+                        else:
+                            start_year = current_year
+
+                    event_start_datetime = datetime(
+                        start_year, datetime.strptime(event_month, "%B").month, event_day
+                    )
+
+            # events are ordered from future to past, so break once we find a past event
+            if event_start_datetime < now:
+                break
+
+            print("Event Start:", event_start_datetime)
+            print("Event End:", event_end_datetime)
+
+            title = event.find("div", class_="PromoSearchResultEvent-title")
```

Comment:
I think here and the below lines might be a good use of a walrus operator to save a line per if-statement. See [here](https://realpython.com/python-walrus-operator/) for examples!